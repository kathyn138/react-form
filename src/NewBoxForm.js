import React, { Component } from 'react';

class NewBoxForm extends Component {
  constructor(props) {
    super(props)
    this.state = { width: '', height: '', backgroundColor: '' };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.addBox(this.state);
    this.setState({ width: '', height: '', backgroundColor: '' });
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="width">Width:</label>
        <input id="width"
          name="width"
          value={this.state.width}
          onChange={this.handleChange} />

        <br></br><label htmlFor="height">Height:</label>
        <input id="height"
          name="height"
          value={this.state.height}
          onChange={this.handleChange} />

        <br></br><label htmlFor="backgroundColor">Background Color:</label>
        <input id="backgroundColor"
          name="backgroundColor"
          value={this.state.backgroundColor}
          onChange={this.handleChange} />
        
        <br></br><button>Add a new box!</button>
      </form>
    );
  }
}

export default NewBoxForm;