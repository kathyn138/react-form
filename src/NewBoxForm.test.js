import React from 'react';
import NewBoxForm from './NewBoxForm';
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";


it('renders without crashing', () => {
  shallow(<NewBoxForm />)
});

it('matches snapshot', function () {
  let wrapper = shallow(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
})

it('allows for changes in height, width, backgroundColor', function () {
  let wrapper = mount(<NewBoxForm />);
  const heightInput = wrapper.find('#height');
  heightInput.instance().value = "10";
  heightInput.simulate("change")

  expect(wrapper.state().height).toEqual("10");

  const widthInput = wrapper.find('#width');
  widthInput.instance().value = "20";
  widthInput.simulate("change");

  expect(wrapper.state().width).toEqual("20");

  const colorInput = wrapper.find('#backgroundColor');
  colorInput.instance().value = "orange";
  colorInput.simulate("change");

  expect(wrapper.state().backgroundColor).toEqual("orange");
});

it('runs a mocked fn on submit', function () {
  const submitFn = jest.fn();
  let wrapper = mount(<NewBoxForm addBox={submitFn} />);
  const form = wrapper.find("form");
  form.simulate("submit");
  expect(submitFn).toHaveBeenCalled();
});

it('resets state on submit', function () {
  const submitFn = jest.fn();
  let wrapper = mount(<NewBoxForm addBox={submitFn} />);

  const heightInput = wrapper.find('#height');
  heightInput.instance().value = "10";
  heightInput.simulate("change")

  expect(wrapper.state().height).toEqual("10");

  const widthInput = wrapper.find('#width');
  widthInput.instance().value = "20";
  widthInput.simulate("change");

  expect(wrapper.state().width).toEqual("20");

  const colorInput = wrapper.find('#backgroundColor');
  colorInput.instance().value = "orange";
  colorInput.simulate("change");

  expect(wrapper.state().backgroundColor).toEqual("orange");

  const form = wrapper.find("form");
  form.simulate("submit");
  expect(wrapper.state()).toEqual({
    height: '',
    width: '',
    backgroundColor: ''
  });
});

