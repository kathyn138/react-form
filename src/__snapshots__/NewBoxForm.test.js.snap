// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="width"
  >
    Width:
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    value=""
  />
  <br />
  <label
    htmlFor="height"
  >
    Height:
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    value=""
  />
  <br />
  <label
    htmlFor="backgroundColor"
  >
    Background Color:
  </label>
  <input
    id="backgroundColor"
    name="backgroundColor"
    onChange={[Function]}
    value=""
  />
  <br />
  <button>
    Add a new box!
  </button>
</form>
`;
